src/contrib/cluster/Akka.Cluster.Tools.Tests/PubSub/DistributedPubSubMediatorSpec.cs

```diff
@@ -0,0 +1,738 @@
+                Receive<string>(input => mediator.Tell(new DistributedPubSubMediator.Publish("content", input.ToUpperInvariant())));
+            }
+        }
+
+        public class Subscriber : UntypedActor
+        {
+            private readonly IActorRef _mediator;
+            private readonly ILoggingAdapter _log;
+
+            public Subscriber()
+            {
+                _log = Context.GetLogger();
+                _mediator = DistributedPubSub.Get(Context.System).Mediator;
+                _mediator.Tell(new DistributedPubSubMediator.Subscribe("content", Self));
+            }
+
+            protected override void OnReceive(object message)
+            {
+                var ack = message as DistributedPubSubMediator.SubscribeAck;
+                if (ack != null && ack.Subscribe.Topic == "content" && ack.Subscribe.Ref.Equals(Self))
+                {
+                    Context.Become(Ready);
+                }
+            }
+
+            private void Ready(object message)
+            {
+                if (message is string) _log.Info("Got {0}", message);
+            }
+        }
+
+        private readonly RoleName _first;
+        private readonly RoleName _second;
+        private readonly RoleName _third;
+
+        private readonly ConcurrentDictionary<string, IActorRef> _chatUsers = new ConcurrentDictionary<string, IActorRef>();
+
+        public DistributedPubSubMediatorSpec() : this(new DistributedPubSubMediatorSpecConfig())
+        {
+        }
+
+        public DistributedPubSubMediatorSpec(DistributedPubSubMediatorSpecConfig config) : base(config)
+        {
+            _first = config.First;
+            _second = config.Second;
+            _third = config.Third;
+        }
+
+        public IActorRef Mediator { get { return DistributedPubSub.Get(Sys).Mediator; } }
+
+        private IActorRef CreateChatUser(string name)
+        {
+            var a = Sys.ActorOf(Props.Create(() => new TestChatUser(Mediator, TestActor)), name);
+            _chatUsers.TryAdd(name, a);
+            return a;
+        }
+
+        private IActorRef ChatUser(string name)
+        {
+            IActorRef a;
+            return _chatUsers.TryGetValue(name, out a) ? a : ActorRefs.Nobody;
+        }
+
+        private void Join(RoleName from, RoleName to)
+        {
+            RunOn(() =>
+            {
+                Cluster.Join(Node(to).Address);
+                CreateMediator();
+            }, from);
+            EnterBarrier(from.Name + "-joined");
+        }
+
+        private void CreateMediator()
+        {
+            var m = DistributedPubSub.Get(Sys).Mediator;
+        }
+
+        private void AwaitCount(int expected)
+        {
+            AwaitAssert(() =>
+            {
+                Mediator.Tell(Count.Instance);
+                Assert.Equal(expected, ExpectMsg<int>());
+            });
+        }
+
+        #endregion
+
+        [MultiNodeFact(Skip = "TODO")]
```

Comment:
From http://getakka.net/docs/testing/multinode-testkit:

If you have multiple phases that need to be executed as part of a test, you can write them like this:

```csharp
[MultiNodeFact]
public void ConvergenceSpecTests()
{
    AClusterOf3MembersMustReachInitialConvergence();
    AClusterOf3MembersMustNotReachConvergenceWhileAnyNodesAreUnreachable();
    AClusterOf3MembersMustNotMoveANewJoiningNodeToUpWhileThereIsNoConvergence();
}
```

This unfortunate design is a byproduct of Xunit and how it recreates the entire test class on each method.